### PR TITLE
Added function state directory

### PR DESCRIPTION
**Pull request by ghsnd**
Added a function state directory that can be used by RMLMapper to keep state. Optional parameters determine the path and the lifetime of the directory.

### CODE REVIEW

File: lib/wrapper.js [hunk 1]
```diff
@@ -399,6 +432,20 @@ class RMLMapperWrapper {
       };
     });
   }
+
+  async _cleanUpStateFolder() {
+    const stateSubFolders = await fsTraditional.promises.readdir(this.stateFolder);
```
Comment (pheyvaer) [inline lib/wrapper.js:437]: Why does the code use `fsTraditional.promises` when that is exactly what `fs-extra` does?

File: lib/wrapper.js [hunk 2]
```diff
@@ -399,6 +432,20 @@ class RMLMapperWrapper {
       };
     });
   }
+
+  async _cleanUpStateFolder() {
```
Comment (pheyvaer) [inline lib/wrapper.js:436]: Add documentation.

**Review by pheyvaer — CHANGES_REQUESTED** [review]
Can you also add a test for this?

File: lib/wrapper.js (hunk 1 above)
Comment (ghsnd) [reply lib/wrapper.js:437]: I remember trying hard to obtain the same using `fs-extra`, but couldn't get it to work. `fsTraditional.promises.readdir(this.stateFolder)` returns a `string[]`, while `fs-extra` uses callbacks to do the processing... Probably my javascript skills are too limited; do you have an idea how to do the same with `fs-extra`?

Comment (pheyvaer) [reply lib/wrapper.js:437]: If I do a local test then `await fsTraditional.promises.readdir` and `await fs.readdir` give the same results.

Comment (ghsnd) [reply lib/wrapper.js:437]: Aah, now I see: if no callback is passed, a promise with the result will be returned :upside_down_face: . See, need to improve my JS knowledge. But it works indeed.

File: lib/wrapper.js (hunk 2 above)
Comment (ghsnd) [reply lib/wrapper.js:442]: done

**Comment by ghsnd** [comment]
@pheyvaer Could you check again? Thanks!

File: index.test.js [hunk 3]
```diff
@@ -236,4 +236,28 @@ describe('Success', function() {
...
+        false,
+        stateFolder,
+        300);
+
+    const rml = fs.readFileSync('./test/functionstate/mapping.ttl', 'utf-8');
+    const sources = {
+      'student.csv': fs.readFileSync('./test/functionstate/student.csv', 'utf-8')
+    };
+
+    await wrapper.execute(rml, {sources, generateMetadata: false, asQuads: true});
```
Comment (pheyvaer) [inline index.test.js:254]: How do we know if the state as actually kept? Don't we need to do at least two executions?

Comment (ghsnd) [reply index.test.js:254]: God point. This just tests if state is kept, not if it is used the next time. I'll add that.

Comment (ghsnd) [reply index.test.js]: Test extended. Could you review again?

**Pull request merged**